src/main/java/com/salesforce/dataloader/security/EncryptionAesUtil.java

```diff
@@ -0,0 +1,195 @@
+            LOGGER.info(customDir + " exists");
+        } else if (customDir.mkdirs()) {
+            LOGGER.info(customDir + " was created");
+        } else {
+            LOGGER.info(customDir + " was not created");
+            throw new RuntimeException("Cannot create directory:" + path);
+        }
+        return Paths.get(path, DEFAULT_KEYFILE_NAME).toString();
+    }
+
+    public String createKeyFile(String filePath) throws IOException {
+        if (filePath == null || filePath.isEmpty()) {
+            filePath = createUserProfileKeyName();
+        }
+        byte[] key = generateEncryptionKey();
+        try (FileOutputStream fos = new FileOutputStream(filePath)) {
+            fos.write(key);
+        }
+        return filePath;
+    }
+
+    public void resetCipherKey(){
+        cipherKey = null;
+    }
+
+    public String setCipherKeyFromFilePath(String filePath) throws IOException, GeneralSecurityException {
+        byte[] data = new byte[1024];
+        try (FileInputStream fis = new FileInputStream(filePath)) {
+            int size = fis.read(data);
+            if (size < ENCRYPTION_KEY_LENGTH_IN_BYTES)
+                throw new GeneralSecurityException("Keyfile content is too short:" + filePath);
+            else
+            {
+                cipherKey = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+                System.arraycopy(data, 0, cipherKey, 0, ENCRYPTION_KEY_LENGTH_IN_BYTES);
+            }
+            return new String(cipherKey, "UTF-8");
+        }
+    }
+
+    public String encryptMsg(String msg) throws GeneralSecurityException {
+        try {
+            return EncryptionUtil.bytesToText(encryptMsg(msg, cipherKey));
+        } catch (Exception e) {
+            LOGGER.error("Fail to encrypt message: " + e.getMessage());
+            throw new  GeneralSecurityException("Error to encrypt message: ", e);
+        }
+    }
+
+    public String decryptMsg(String cipherMsgString) throws GeneralSecurityException {
+        try {
+            byte[] cipherMsg = EncryptionUtil.textToBytes(cipherMsgString);
+            return decryptMsg(cipherMsg, cipherKey);
+        } catch (Exception e) {
+            LOGGER.error("Fail to decrypt message: " + e.getMessage());
+            throw new GeneralSecurityException("Error to encrypt message: ", e);
+        }
+    }
+
+    synchronized public byte[] encryptMsg(String msg, byte[] encryptionKey) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException, InvalidParameterSpecException, UnsupportedEncodingException, BadPaddingException, IllegalBlockSizeException {
+
+        if(encryptionKey == null)
+            return msg.getBytes("UTF-8");
+        SecretKeySpec key = new SecretKeySpec(encryptionKey, "AES");
+        cipher.init(Cipher.ENCRYPT_MODE, key);
+
+        byte[] cipherText = cipher.doFinal(msg.getBytes("UTF-8"));
+        byte[] iv = cipher.getIV();
+        return concatenateByteArray(iv, cipherText);
```

Comment:
The length of the automatically-generated IV doesn't seem to be well-defined anywhere. This means that different versions of the JRE, or even just different crypto providers (Java's crypto API is pluggable), may return different generated IV lengths. Currently, this class assumes that the IV length is the same as the key length, but this is by no means guaranteed. A custom IV (use SecureRandom to generate it, same as the key, but *unique for every message*) of a known length would avoid this risk.